### DISCUSSION TITLE
Getting "dns error" on application startup when deployed on EC2 using docker container

### DISCUSSION THREAD

**Discussion by Star3Lord** [awslabs/aws-sdk-rust#782] [Q&A]
I am unable to make request using rust application on docker container deployed to EC2.
It works perfectly on localhost but when I deploy it, I get DisptachFailure with dns error.
I enabled tracing to see if endpoint url was maybe incorrect but the request was identical to the one on my computer except of os. I have all incoming and outgoing ports open for debugging.
I have a reverse proxy in the front but I don't think it has anything to do with this as I never had any problem making outbound request from application inside the containers before, but I am not 100%.

The error
```
Error: Unhandled(Unhandled { source: DispatchFailure(DispatchFailure { source: ConnectorError 
// { kind: Io, source: hyper::Error(Connect, ConnectError("dns error", Custom { kind: Uncategorized, error: "failed to lookup address information: 
// Name or service not known" })) } }) })
```

I am using the code below
```rs
use aws_sdk_s3 as s3;


let region_provider = aws_config::meta::region::RegionProviderChain::first_try(s3::config::Region::new(config.s3_region_name.to_string()));
    let region = region_provider.region().await.unwrap();

    let aws_config = aws_config::from_env().load().await;

    let credentials = s3::config::Credentials::new(
        config.s3_access_key_id.to_string(),
        config.s3_secret_access_key.to_string(),
        None,
        None,
        "my_creds",
    );

    let s3_config = s3::config::Builder::from(&aws_config)
        .retry_config(aws_config::retry::RetryConfig::standard())
        .region(region)
        .endpoint_url(config.s3_endpoint_url.to_string())
        .credentials_provider(credentials)
        .build();
    let s3_client = s3::Client::from_conf(s3_config);
```
    
Is there something that I am overlooking?
I can't use any S3 services in my setup because of this issue.

**Comment by rcoh** [comment]
can you exec into your docker container and see if `curl`ing the URL works?

**Comment by mzygQAQ** [comment]
try add s3_config.force_path_style(true)

**Comment by randomseed42** [comment]
Similar error happend here as well on my EC2, I have a rust program working in Ubuntu EC2 to deal with s3 files, `ulimit -n = 1024`, probably I need to increase max open files limit?

```rust
Error: DispatchFailure(
  DispatchFailure { 
    source: ConnectorError { 
      kind: Io, 
      source: hyper::Error(
        Connect, ConnectError(
          "dns error", 
          Os {
             code: 24, kind: Uncategorized, message: "Too many open files"
          }
        )
      ), 
      connection: Unknown 
    } 
  }
)
```
